### ISSUE TITLE
[Machine Learning] Hyperparameter Tuning in Decision Trees and Random Forests with Python

### ISSUE THREAD

**Issue by himani007** [section-engineering-education/engineering-education#5569]
### Topic Suggestion
[Machine Learning] Hyperparameter Tuning in Decision Trees and Random Forests using Python

#### Proposed title of the article
Hyperparameter Tuning in Decision Trees and Random Forests

#### Proposed article introduction
[Machine Learning] Introduction to ML

After years of hard work, we've reached a stage where we are using computers to analyze millions of data points and provide insights that even the human eye couldn't catch. But our machine Learning Model is only as good as its accuracy on unseen data, i.e how well our model generalizes. In this article, I'll solve a Binary Classification Problem, using a Decision Tree Classifier and Random forests and solve the overfitting problem by tuning their Hyperparameter and compare results. 
This shall be a beginner-friendly article for those new to Decision Trees.  

#### Key takeaways 
- Introduction to ML
- Elaborated Introduction to  Decision Trees and Random Forests
- Gini Index and Feature Importance
- Overfitting and Regularisation 
- Ensembling
- Tuning Hyperparameters: `max_depth`, `max_leaf_nodes`, `n_estimators` etc
 

#### Article quality
This is a unique article from the published articles on the section.io/engineering-education. Understanding the concept of overfitting and learning how to overcome it is as important as preprocessing the data and making it fir for modeling. 
Similarly, tuning the decision tree hyperparameters is as important as learning how to use this approach in ML, which will also be covered in-depth in this article.

**Comment by ahmadmardeni1** [comment]
Sounds like a helpful topic - let's please be sure it adds value beyond what is in any official docs and/or what is covered in other blog sites. (the articles should go beyond a basic explanation - and it is always best to reference any EngEd article and build upon it). @himani007 

Please be attentive to grammar/readability and make sure that you put your article through a thorough editing review prior to submitting it for final approval. (There are some great free tools that we reference in EngEd resources.)
ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.

Please be sure to double-check that it does not overlap with any existing EngEd articles, articles on other blog sites, or any incoming EngEd topic suggestions (if you haven't already) to avoid any potential article closure, please reference any relevant EngEd articles in yours. - Approved
